design-proposals/vm-delete-protection.md

```diff
@@ -0,0 +1,155 @@
+# Overview
+
+VirtualMachine Delete Protection prevents the accidental deletion of VMs from a cluster. As long as deletion protection
+is activated for certain VM objects, they cannot be deleted unless the protection is intentionally removed.
+
+## Motivation
+
+KubeVirt clusters are often managed by automations, CLI commands, 3rd party tools or UI actions, for example to deploy,
+reconfigure or delete VMs. These automations may result in accidental deletion of VMs that should have not been deleted.
+Affected VMs could be critical to application/s the cluster is running, or they may contain crucial data. Therefore, the
+VM deletion protection aims to avoid accidental degradation of applications or loss of data.
```

Comment:
I'd remove `they may contain crucial data` and `or loss of data` from here. We don't want to have any reference to storage, since it may raise questions about the protection of PVC from deletion.

Our motivation is simple: Protect VMs from deletion that may run business critical applications or services, that deletion of them could cause out of service (e.g. SQL server is not available to handle requests).

Reply:
I think it's valid to mention this. If a PVC is deleted as a result of a cascaded delete that would result in loss of data.

Reply:
Thanks for your inputs. Based on your comments, I've improved the text.
I find out Felix's argument valid, so I will keep the mention to storage.